lib/src/native/controller/controller.dart

```diff
@@ -377,4 +377,41 @@ class NativeAdController extends LoadShowAd<NativeAdEvent>
       assert(!reason.isNegative, 'You must specify a valid reason');
     return channel.invokeMethod('muteAd', {'reason': reason});
   }
+
+  Future<String?> getHeadline() async {
```

Comment:
It'd be better to make it sync. That means you'd get the ad info on the ad load. Something similar to what I did on `onAdLoaded` above.

Reply:
Seems to work just by making it sync. Can you explain more? Did you mean like what you did on `muteThisAd` ? like calling `ensureAdNotDisposed()` before invoking the method?

Reply:
Similar to what I did here

https://github.com/bdlukaa/native_admob_flutter/blob/master/lib/src/native/controller/controller.dart#L127

and here

https://github.com/bdlukaa/native_admob_flutter/blob/master/lib/src/native/controller/controller.dart#L300

When the ad is loaded, the platform side sends more info about the native ad.

Reply:
Hi, I implemented these but there seems to be one disadvantage. It seems that in your plugin, every NativeAd widget has its own controller, either we have attached one to it or not. This way, we would store unnecessary values (headline, body, images, ...) in memory for ads which the user has not attached any controller to them, so (s)he won't even have any access to these variables. Please correct me if I'm wrong.

Reply:
I guess that's a little price to pay in order to have the best dev experience